OpenRA.Game/Actor.cs

```diff
@@ -132,10 +132,11 @@ internal Actor(World world, string name, TypeDictionary initDict)
 		Rectangle DetermineBounds()
 		{
 			var si = Info.TraitInfoOrDefault<SelectableInfo>();
+			var autoSelectionSize = TraitsImplementing<IAutoSelectionSize>();
 			var size = (si != null && si.Bounds != null) ? new int2(si.Bounds[0], si.Bounds[1]) :
-				TraitsImplementing<IAutoSelectionSize>().Select(x => x.SelectionSize(this)).FirstOrDefault();
+				autoSelectionSize.Select(x => x.SelectionSize(this)).FirstOrDefault();
 
-			var offset = -size / 2;
+			var offset = autoSelectionSize.Select(x => x.SelectionOffset(this)).FirstOrDefault() - size / 2;
```

Comment:
This introduces an inconsistency: `Selectable` *overrides* the size, but *combines* the offsets.  It would be more consistent to override in both.

Reply:
We will also need to mirror this behaviour with the visual sizes wrt #13899.

Reply:
I am probably not getting this right, but your suggestion would break the offsets:
![image](https://user-images.githubusercontent.com/756669/30522562-756cee32-9bd2-11e7-8ccc-c2fc7e6feeeb.png)

Reply:
Assuming this is what you meant:
```diff
@@ -133,12 +133,12 @@ Rectangle DetermineBounds()
 		{
 			var si = Info.TraitInfoOrDefault<SelectableInfo>();
 			var autoSelectionSize = TraitsImplementing<IAutoSelectionSize>();
-			var size = (si != null && si.Bounds != null) ? new int2(si.Bounds[0], si.Bounds[1]) :
+
+			var size = (si != null && si.Bounds != null && si.Bounds.Length >= 2) ? new int2(si.Bounds[0], si.Bounds[1]) :
 				autoSelectionSize.Select(x => x.SelectionSize(this)).FirstOrDefault();
 
-			var offset = autoSelectionSize.Select(x => x.SelectionOffset(this)).FirstOrDefault() - size / 2;
-			if (si != null && si.Bounds != null && si.Bounds.Length > 2)
-				offset += new int2(si.Bounds[2], si.Bounds[3]);
+			var offset = (si != null && si.Bounds != null && si.Bounds.Length > 2) ? new int2(si.Bounds[2], si.Bounds[3]) :
+				autoSelectionSize.Select(x => x.SelectionOffset(this)).FirstOrDefault() - size / 2;
 
 			return new Rectangle(offset.X, offset.Y, size.X, size.Y);
 		}
```

Reply:
Ah, hm, I think you need `new int2(si.Bounds[2], si.Bounds[3]) - size / 2` instead of just `new int2(si.Bounds[2], si.Bounds[3])`. Before these changes it always created a `offset` variable with the value of `- size / 2` and then added a `new int2(si.Bounds[2], si.Bounds[3])` on top if the condition was met.

(Or just use more parenthesis:
```
var offset = ((si != null && si.Bounds != null && si.Bounds.Length > 2) ? new int2(si.Bounds[2], si.Bounds[3]) :
	autoSelectionSize.Select(x => x.SelectionOffset(this)).FirstOrDefault()) - size / 2;
```
... ^^)